### Bayes Theorem

The cigar example in the lab illustrates the application of Bayes' theorem with its calculation using the formula. Unfortunately, that calculation is complicated and can cause confusion and/or incorrect substitution of the involved
probability values. Fortunately, here is another approach that is much more intuitive and easier:

Assume some convenient value for the total of all items involved, then
construct a table of rows and columns with the individual cell frequencies
based on the known probabilities.

For example, let's assume that the adult population in Boone county is 100,000. Now we can use the given information to create a table.

*Number of males who smoke cigars:* 51% of adults are males; so there are 51,000 males. If 9.5% of them smoke, that makes 0.095 x 51,000 = 4845. Then, males who do not smoke are 51,000 - 4845 = 46,155. See the table where these values go.


*Number of females who smoke cigars:* 49% of the adults are females, that makes 49,000. 1.7% of them are smokers, so 0.017 x 49,000 = 833. The number of females who do not smoke is 49,000 - 833 = 48,167. Again look at the table below. 

In [1]:
cigar <- matrix(c(4845, 833, 46155, 48167), ncol=2)
colnames(cigar) <- c('smoker','nonsmoker')
rownames(cigar) <- c('male','female')
cigar.table <- as.table(cigar)

addmargins(cigar)

smoker nonsmoker Sum   
male   4845   46155      51000
female  833   48167      49000
Sum    5678   94322     100000

The above table involves simple arithmetic. Simply partition the
assumed population into the different cell categories by finding suitable percentages.

Now we can easily address the key question as follows: To find the probability of
getting a male subject, given that the subject smokes cigars, simply use the same
conditional probability described before. 

To find the probability of getting a
male given that the subject smokes, restrict the table to the column of cigar smokers, then
find the probability of getting a male in that column. Among the 5678 cigar smokers,
there are 4845 males, so the probability we seek is 4845/5678 = 0.85329341. That is,
$P(M | C)$ = 4845/5678 = 0.85329341 = 0.853 (rounded).

**Activity 1:** Now, your turn: the actual population of Boone county is 170,733 (as of 2013). Create the above table with actual population values for the given percentages and find the actual $P(M | C)$.

In [ ]:
#solution goes here.

a) Now, using the same table, randomly select an individual, what is the prior probability that the selected person is a female?

b) You later learn that the randomly selected person was smoking a cigar. Use this additional information to find the posterior 
probability that the selected person is a female.

In [ ]:
addmargins(prop.table(cigar))

In [ ]:
# solution goes here.

Load the framingham data from the directory '/datasets/framingham'.

In [ ]:
framingham_data <- read.csv("../../../datasets/framingham/framingham.csv")
head(framingham_data)

**Activity 2:** Create a two-way table from this data set with diabetes condition in the columns and gender in the rows. Use addmargins to add totals.


In [ ]:
dia <- with(framingham_data,table(<what goes here>,<what goes here>))
colnames(dia) <- c('nondiabetes','diabetes')
rownames(dia) <- c('female','male')
<what goes here>(dia)

**Activity 3: **What is the probability for a female to have diabetes? Let d be an event of diabetes and d' be event of nondiabetes. Similarly let g be the event of male and g' be event of female. Find $P(d | g')$ using Bayes formula.

            
                    p(d) * p(g'/d)
     p(d/g') =  -------------------------------------
               [p(d) * p(g'/d)] + [ p(d') * p(g'/d')]
            

In [ ]:
#solution goes here.
